### PR TITLE
Ry/game socket

### LINKED ISSUE #70: Create game Websocket
Implement communication between FE and BE using websockets. Gateway for the backend and component for the frontend. Rememebr types, vue js quite hard to find clear examples with types. RTFM ;)

What is the minimum information is required to be an instance of "game state"

### PR DESCRIPTION
**Pull request by Ryno95**
Basic socket setup. Game Elements are now rendered by using the data sent from the backend.

closes  #70

### CODE REVIEW

File: backend/src/pong/pong.gateway.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	WebSocketGateway,
+	WebSocketServer,
+} from '@nestjs/websockets';
+import { Server, Socket } from 'socket.io';
+import { Ball, Paddle } from './Game.types';
+
+@WebSocketGateway({
+	namespace: '/pong',
+	cors: {
+		origin: '*:*',
```
Comment (vincentvis) [inline backend/src/pong/pong.gateway.ts:14]: Why `*:*` and not just `*`?
I can't find anything about what the extra `:*` would do.

**Review by vincentvis — APPROVED** [review]
I feel like quite a lot has changed already from what I've seen for further development of the game :) So I guess this is just the in between step that we hadn't merged yet..

File: backend/src/pong/pong.gateway.ts (hunk 1 above)
Comment (Ryno95) [reply backend/src/pong/pong.gateway.ts:14]: Honestly, I have no idea anymore, it was something from a CORS issue on StackOF , but cant remember what is was, was put here out of habit. changed it to "*"

File: backend/src/pong/pong.gateway.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+	async handleConnection(socket: Socket) {
+		console.log('\n!Socket is connected!\n');
+		this.sendHallo('Hallo frontend!!!');
+		this.sendPaddlePosition(this.playerOnePaddle);
+		this.sendPaddlePosition(this.playerTwoPaddle);
+		this.sendBallPosition(this.ball);
+	}
+
+	@SubscribeMessage('hallo')
```
[the remark this answers is not in the document]
Comment (Ryno95) [reply backend/src/pong/pong.gateway.ts:60]: indeed, removed it form the following branch

**Pull request merged**